skbio/sequence/_sequence.py

```diff
@@ -538,9 +586,13 @@ def _assert_can_cast_to(cls, target):
     def _positional_metadata_axis_len_(self):
         return len(self)
 
+    @overrides(IntervalMetadataMixin)
+    def _interval_metadata_axis_len_(self):
+        return len(self)
+
     @stable(as_of="0.4.0")
     def __init__(self, sequence, metadata=None, positional_metadata=None,
-                 lowercase=False):
+                 interval_metadata=None, lowercase=False):
```

Comment:
@gregcaporaso @jairideout @ebolyen, flagging this comment for comment :) Basically, we're unsure about subtleties of the implications of API stability